### PR TITLE
BOAC-95, cohort member listing gets order_by, offset and limit params

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-95

Backend/front-end support for sorting and paging of cohort member lists (and search results). Includes initial formatting on front-end.

### CODE REVIEW

File: boac/models/cohort_filter.py [hunk 1]
```diff
@@ -38,35 +41,92 @@ def create(cls, label, team_codes, uid):
...
+                member_csid,
+                asc_sport,
+                code,
+                member_uid
+            FROM team_members
+            WHERE code IN ({team_codes})
+            ORDER BY {order_by}
+            OFFSET {offset}
+            LIMIT {limit}
+        """.format(team_codes=comma_separated_codes, order_by=order_by, offset=offset, limit=limit))
```
Comment (johncrossman) [inline boac/models/cohort_filter.py]: I believe raw SQL is only option here. The team_codes live in `filter_criteria` JSON, in cohorts table.

File: boac/static/app/cohort/cohort.html [hunk 2]
```diff
@@ -1,67 +1,79 @@
...
-          <option value="name">Name</option>
-          <option value="uid">UID</option>
-        </select>
-      </div>
+  <div class="flex-container cohort-manage-columns" data-ng-show="!error && cohort">
+    <div class="cohort-column-search">
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
+      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
+      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
+      Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
```
Comment (johncrossman) [inline boac/static/app/cohort/cohort.html]: Placeholder for search form.

File: boac/api/cohort_controller.py [hunk 3]
```diff
@@ -88,75 +88,37 @@ def delete_cohort(cohort_id):
...
+def load_member_profiles(team):
+    for member in team['members']:
+        uid = member['uid']
+        cache_key = 'user/{uid}'.format(uid=uid)
+        canvas_profile = app.cache.get(cache_key) if app.cache else None
+        if not canvas_profile:
+            canvas_profile = canvas.get_user_for_uid(uid)
+            # Cache Canvas profiles
+            if app.cache and canvas_profile:
+                app.cache.set(cache_key, canvas_profile)
```
Comment (johncrossman) [inline boac/api/cohort_controller.py]: Caching is now per canvas_profile. We do not cache teams.

**Comment by raydavis** [comment]
Looks good to me, @johncrossman -- just waiting for SQLAlchemy & Travis to reach an understanding...

**Comment by johncrossman** [comment]
This query might be possible without raw SQL. I'll give it a shot.  cc: @raydavis

**Comment by raydavis** [comment]
FWIW, @johncrossman , I've hit these issues in the wild when I tried to do any SQL outside the sacred bounds of a Flask request context (since that's the assumed transaction boundary) -- in other words, the issue was the call stack, not the nature of the SQL query.

**Comment by johncrossman** [comment]
Operation 'remove raw SQL' was a success. Finally ready for proper review.  cc: @raydavis

**Pull request merged**